src/app/Library/CrudPanel/Traits/Validation.php

```diff
@@ -397,4 +351,83 @@ private function getValidationRulesAndMessagesFromField($field, $parent = false)
 
         return [$rules, $messages];
     }
+
+    /**
+     * Return an array containing the request rules and the field/controller rules merged.
+     * The rules in request will take precedence over the ones in controller/fields.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  array  $rules
+     * @return array
+     */
+    private function mergeRules($request, $rules)
```

Comment:
Note to self: this hasn't changed at all, just moved in a different place in the same file.